src/borg/helpers/fs.py

```diff
@@ -19,6 +21,27 @@
 py_37_plus = sys.version_info >= (3, 7)
 
 
+def ensure_dir(path, mode=stat.S_IRWXU, pretty_deadly=True):
+    """
+    Ensures that the dir exists with the right permissions.
+    1) Make sure the directory exists in a race-free operation
+    2) If mode is not None and the directory has been created, give the right
```

Comment:
according to the python docs, it only assures that the **leaf directory** will get these permissions.